### PR TITLE
Track verified callbacks

### PR DESCRIPTION
**Pull request by nickpatrick**
iOS equivalent of https://github.com/radarlabs/radar-sdk-android/pull/362

- Remove `trackVerifiedToken()` and change `trackVerified()` to return `RadarVerifiedLocationToken`
- Remove `token` param from `startTrackingVerified()` and change `RadarVerifiedDelegate` to return `RadarVerifiedLocationToken`
- Make `token.expiresAt` override `interval` if needed
- Add `getVerifiedLocationToken()` which will return last token if not expired and request fresh token if needed
- Random logging and comment cleanup: log format consistency, fix typos, remove unnecessary/obvious comments, lowercase one-line comments

### CODE REVIEW

**Comment by david-goodfellow** [comment]
some failing tests:

`test_Radar_getContext_errorLocation, Asynchronous wait failed: Exceeded timeout of 60 seconds, with unfulfilled expectations: "callback". /Users/runner/work/radar-sdk-ios/radar-sdk-ios/RadarSDKTests/RadarSDKTests.m:826`
  ```

    [self waitForExpectationsWithTimeout:60
                                 handler:^(NSError *_Nullable error) {
  ```

 `test_Radar_getContext_errorLocation, failed /Users/runner/work/radar-sdk-ios/radar-sdk-ios/RadarSDKTests/RadarSDKTests.m:829`
  ```
                                     if (error) {
                                         XCTFail();
                                     }
  ```

  `test_Radar_getContext_success, ((status) equal to (RadarStatusErrorLocation)) failed: ("0") is not equal to ("3") /Users/runner/work/radar-sdk-ios/radar-sdk-ios/RadarSDKTests/RadarSDKTests.m:821`
  ```
    [Radar getContextWithCompletionHandler:^(RadarStatus status, CLLocation *_Nullable location, RadarContext *_Nullable context) {
        XCTAssertEqual(status, RadarStatusErrorLocation);

  ```

File: RadarSDK/RadarLocationManager.m [hunk 1]
```diff
@@ -519,7 +520,7 @@ - (void)replaceSyncedGeofences:(NSArray<RadarGeofence *> *)geofences {
                     NSString *notificationRepeats = [geofence.metadata objectForKey:@"radar:notificationRepeats"];
                     if (notificationRepeats) {
                         repeats = [notificationRepeats boolValue];
-                        [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:[NSString stringWithFormat:@"repeats is bool vaue: %d", repeats]];
+                        [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:[NSString stringWithFormat:@"Notification repeats | repeats = %d", repeats]];
```
Comment (lmeier) [inline RadarSDK/RadarLocationManager.m:523]: This copy will be confusing in debug logs, as it'll appear as "Notification repeats" in the main log line, only showing "repeats = false" once you click in.

File: RadarSDK/RadarVerificationManager.m [hunk 2]
```diff
@@ -31,8 +31,12 @@
 
 @interface RadarVerificationManager ()
 
-@property (strong, nonatomic) NSTimer *timer;
+@property (assign, nonatomic) BOOL started;
+@property (strong, nonatomic) NSTimer *intervalTimer;
 @property (nonatomic, retain) nw_path_monitor_t monitor;
+@property (strong, nonatomic) RadarVerifiedLocationToken *lastToken;
+@property (assign, nonatomic) NSTimeInterval lastTokenSystemUptime;
+@property (assign, nonatomic) BOOL lastTokenBeacons;
```
Comment (lmeier) [inline RadarSDK/RadarVerificationManager.m:39]: Nit: I think `latest` for these 3 might be slightly better verbiage here. "last" to me first parses as the final element in a list and sorta sounds like it's static / won't change.

File: RadarSDK/RadarVerifiedLocationToken.m [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    
+    return nil;
+}
+
+- (NSDictionary *)dictionaryValue {
+    NSMutableDictionary *dict = [NSMutableDictionary new];
+    dict[@"user"] = [self.user dictionaryValue];
+    dict[@"events"] = [RadarEvent arrayForEvents:self.events];
+    dict[@"token"] = self.token;
+    dict[@"expiresAt"] = [RadarUtils.isoDateFormatter stringFromDate:self.expiresAt];
```
Comment (lmeier) [inline RadarSDK/RadarVerifiedLocationToken.m:91]: Do we want a null check as `stringFromDate` expects a non-nullable `NSDate`?

Also nit, but I slightly prefer:
`[[RadarUtils isoDateFormatter] stringFromDate:self.expiresAt];`

as it matches our more standard objective-c style.

File: RadarSDK/RadarVerificationManager.m [hunk 4]
```diff
@@ -153,105 +162,28 @@ - (void)trackVerifiedWithBeacons:(BOOL)beacons completionHandler:(RadarTrackComp
...
-                
-                return;
+- (void)startTrackingVerifiedWithInterval:(NSTimeInterval)interval beacons:(BOOL)beacons {
+    self.started = YES;
+    
+    __block void (^trackVerified)(void);
+    __block __weak void (^weakTrackVerified)(void);
+    trackVerified = ^{
+        weakTrackVerified = trackVerified;
+
```
Comment (lmeier) [inline RadarSDK/RadarVerificationManager.m:172]: This pattern of using a `weak` version is interesting (I wanna read more up on this)

**Review by lmeier — COMMENTED** [review]
looking good, basically just nit style stuff. Wonder why the tests aren't passing —potentially merged against an older `master`?

File: RadarSDK/RadarVerificationManager.m (hunk 4 above)
Comment (nickpatrick) [reply RadarSDK/RadarVerificationManager.m:172]: Same pattern as `mockTracking`, still need to test here https://github.com/radarlabs/radar-sdk-ios/blob/master/RadarSDK/Radar.m#L324

File: RadarSDK/RadarVerifiedLocationToken.m (hunk 3 above)
Comment (nickpatrick) [reply RadarSDK/RadarVerifiedLocationToken.m:91]: Could go either way here. We are using dot notation for `self.expiresAt` in the same line

File: RadarSDK/RadarVerificationManager.m (hunk 2 above)
Comment (nickpatrick) [reply RadarSDK/RadarVerificationManager.m:39]: Could go either way here. fwiw we use `lastLocation`, `lastMovedAt`, etc. in `RadarState`, was following that naming convention

**Pull request merged**